app/src/main/kotlin/com/softteco/template/ui/feature/bluetooth/BluetoothScreen.kt

```diff
@@ -0,0 +1,136 @@
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.Button
+import androidx.compose.material3.Card
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.softteco.template.R
+import com.softteco.template.data.bluetooth.BluetoothDevice
+import com.softteco.template.ui.components.CustomTopAppBar
+import com.softteco.template.ui.components.TextSnackbarContainer
+
+@Composable
+fun BluetoothScreen(
+    modifier: Modifier = Modifier,
+    viewModel: BluetoothViewModel = hiltViewModel(),
+    onBackClicked: () -> Unit = {}
+) {
+    val state by viewModel.state.collectAsState()
+
+    ScreenContent(
+        state,
+        modifier,
+        onBackClicked
+    )
+}
+
+@Composable
+private fun ScreenContent(
+    state: BluetoothViewModel.State,
+    modifier: Modifier = Modifier,
+    onBackClicked: () -> Unit = {}
+) {
+    TextSnackbarContainer(
+        modifier = modifier,
+        snackbarText = stringResource(state.snackBar.textId),
+        showSnackbar = state.snackBar.show,
+        onDismissSnackbar = state.dismissSnackBar,
+    ) {
+        Column(modifier = Modifier.fillMaxWidth()) {
+            CustomTopAppBar(
+                stringResource(id = R.string.bluetooth),
+                showBackIcon = true,
+                modifier = Modifier.fillMaxWidth(),
+                onBackClicked = onBackClicked
+            )
+            BluetoothDevicesList(state.bluetoothDevices)
+        }
+    }
+}
+
+@Composable
+fun BluetoothDevicesList(bluetoothDevices: MutableList<BluetoothDevice>) {
+    LazyColumn(
+        modifier = Modifier.fillMaxWidth(),
+        contentPadding = PaddingValues(16.dp)
+    ) {
+        items(bluetoothDevices) { bluetoothDevice ->
+            BluetoothDeviceCard(
+                bluetoothDevice.name,
+                bluetoothDevice.macAddress,
+                bluetoothDevice.rssi
+            )
+        }
+    }
+}
+
+@Composable
+fun BluetoothDeviceCard(name: String, macAddress: String, rssi: Int) {
+    Card(
+        modifier = Modifier
+            .padding(10.dp)
+            .fillMaxWidth()
+            .wrapContentHeight(),
+        shape = MaterialTheme.shapes.medium
+    ) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+            Image(
+                painter = painterResource(R.drawable.ic_bluetooth),
+                contentDescription = null,
```

Comment:
please add a description, because you will receive a warning in the play console about this